service/dap/server.go

```diff
@@ -2375,7 +2393,13 @@ func (s *Server) onExceptionInfoRequest(request *dap.ExceptionInfoRequest) {
 		if err == nil {
 			var buf bytes.Buffer
 			fmt.Fprintln(&buf, "Stack:")
-			terminal.PrintStack(s.toClientPath, &buf, apiFrames, "\t", false)
+			userLoc := g.UserCurrent()
+			userFuncPkg := fnPackageName(&userLoc)
+			terminal.PrintStack(s.toClientPath, &buf, apiFrames, "\t", false, func(s api.Stackframe) bool {
+				// Include all stack frames if the stack trace is for a system goroutine,
+				// otherwise, skip runtime stack frames.
+				return userFuncPkg == "runtime" || !strings.HasPrefix(s.Location.Function.Name(), "runtime")
```

Comment:
s.Location.Function could be nil. Also it should be `"runtime."` so that it doesn't elide cgo frames.

Reply:
done.